Failing optimistic registry tests

Looks like our recent changes (#214 & #71) broke a few tests:

```
161 passing (3m)
  6 failing

  1) Optimistic recipient registry
       managing recipients
         allows anyone to submit registration request:
     Uncaught RuntimeError: abort(AssertionError: Expected "RequestSubmitted" event to have 5 argument(s), but it has 6). Build with -s ASSERTIONS=1 for more info.
      at process.abort (node_modules/@ethereum-waffle/compiler/node_modules/solc/soljson.js:1:14382)
      at process.emit (node:events:378:20)
      at process.emit (node_modules/source-map-support/source-map-support.js:485:21)
      at process.emit (node_modules/ganache-core/node_modules/source-map-support/source-map-support.js:465:21)
      at emit (node:internal/process/promises:135:22)
      at processPromiseRejections (node:internal/process/promises:241:25)
      at processTicksAndRejections (node:internal/process/task_queues:95:32)
      at runNextTicks (node:internal/process/task_queues:63:3)
      at listOnTimeout (node:internal/timers:526:9)
      at processTimers (node:internal/timers:500:7)

  2) Optimistic recipient registry
       managing recipients
         allows anyone to submit registration request:
     Error: done() called multiple times in test <Optimistic recipient registry managing recipients allows anyone to submit registration request> of file /Users/samrichards/code/ef/clrfund/contracts/tests/recipientRegistry.ts
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:94:5)

  3) Optimistic recipient registry
       managing recipients
         allows anyone to submit removal request:
     Error: VM Exception while processing transaction: revert Ownable: caller is not the owner
      at OptimisticRecipientRegistry.onlyOwner (@openzeppelin/contracts/access/Ownable.sol:43)
      at OptimisticRecipientRegistry.removeRecipient (contracts/recipientRegistry/OptimisticRecipientRegistry.sol:127)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:94:5)
      at EthModule._sendTransactionAndReturnHash (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:1332:9)
      at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:102:18)
      at WaffleMockProviderAdapter.send (node_modules/@nomiclabs/hardhat-waffle/src/waffle-provider-adapter.ts:31:20)

  4) Optimistic recipient registry
       managing recipients
         should not accept removal request with incorrect deposit size:

      AssertionError: Expected transaction to be reverted
      + expected - actual

      -Transaction NOT reverted.
      +Transaction reverted.



  5) Optimistic recipient registry
       managing recipients
         allows owner to challenge removal request:
     Error: VM Exception while processing transaction: revert Ownable: caller is not the owner
      at OptimisticRecipientRegistry.onlyOwner (@openzeppelin/contracts/access/Ownable.sol:43)
      at OptimisticRecipientRegistry.removeRecipient (contracts/recipientRegistry/OptimisticRecipientRegistry.sol:127)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:94:5)
      at EthModule._sendTransactionAndReturnHash (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:1332:9)
      at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:102:18)
      at WaffleMockProviderAdapter.send (node_modules/@nomiclabs/hardhat-waffle/src/waffle-provider-adapter.ts:31:20)

  6) Optimistic recipient registry
       managing recipients
         allows anyone to execute unchallenged removal request:
     Error: VM Exception while processing transaction: revert Ownable: caller is not the owner
      at OptimisticRecipientRegistry.onlyOwner (@openzeppelin/contracts/access/Ownable.sol:43)
      at OptimisticRecipientRegistry.removeRecipient (contracts/recipientRegistry/OptimisticRecipientRegistry.sol:127)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:94:5)
      at EthModule._sendTransactionAndReturnHash (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:1332:9)
      at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:102:18)
      at WaffleMockProviderAdapter.send (node_modules/@nomiclabs/hardhat-waffle/src/waffle-provider-adapter.ts:31:20)

```

--- a/contracts/tests/recipientRegistry.ts
+++ b/contracts/tests/recipientRegistry.ts
@@ -598,7 +598,8 @@ describe('Optimistic recipient registry', () => {
           RequestType.Registration,
           recipientAddress,
           metadata,
-          currentTime
+          currentTime,
+          requester.address
         )
       expect(await provider.getBalance(registry.address)).to.equal(baseDeposit)
     })
@@ -827,16 +828,14 @@ describe('Optimistic recipient registry', () => {
       }
     })
 
-    it('allows anyone to submit removal request', async () => {
+    it('allows owner to submit removal request', async () => {
       await registry.addRecipient(recipientAddress, metadata, {
         value: baseDeposit,
       })
       await provider.send('evm_increaseTime', [86400])
       await registry.executeRequest(recipientId)
 
-      const requestSubmitted = await registry
-        .connect(requester)
-        .removeRecipient(recipientId, { value: baseDeposit })
+      const requestSubmitted = await registry.removeRecipient(recipientId)
       const currentTime = await getCurrentTime()
       expect(requestSubmitted)
         .to.emit(registry, 'RequestSubmitted')
@@ -845,15 +844,25 @@ describe('Optimistic recipient registry', () => {
           RequestType.Removal,
           ZERO_ADDRESS,
           '',
-          currentTime
+          currentTime,
+          deployer.address
         )
-      expect(await provider.getBalance(registry.address)).to.equal(baseDeposit)
+    })
+
+    it('allows only owner to submit removal request', async () => {
+      await registry.addRecipient(recipientAddress, metadata, {
+        value: baseDeposit,
+      })
+      await provider.send('evm_increaseTime', [86400])
+      await registry.executeRequest(recipientId)
+
+      await expect(
+        registry.connect(requester).removeRecipient(recipientId)
+      ).to.be.revertedWith('Ownable: caller is not the owner')
     })
 
     it('should not accept removal request if recipient is not in registry', async () => {
-      await expect(
-        registry.removeRecipient(recipientId, { value: baseDeposit })
-      ).to.be.revertedWith(
+      await expect(registry.removeRecipient(recipientId)).to.be.revertedWith(
         'RecipientRegistry: Recipient is not in the registry'
       )
     })
@@ -865,13 +874,13 @@ describe('Optimistic recipient registry', () => {
       await provider.send('evm_increaseTime', [86400])
       await registry.executeRequest(recipientId)
 
-      await registry.removeRecipient(recipientId, { value: baseDeposit })
+      await registry.removeRecipient(recipientId)
       await provider.send('evm_increaseTime', [86400])
       await registry.connect(requester).executeRequest(recipientId)
 
-      await expect(
-        registry.removeRecipient(recipientId, { value: baseDeposit })
-      ).to.be.revertedWith('RecipientRegistry: Recipient already removed')
+      await expect(registry.removeRecipient(recipientId)).to.be.revertedWith(
+        'RecipientRegistry: Recipient already removed'
+      )
     })
 
     it('should not accept new removal request if previous removal request is not resolved', async () => {
@@ -881,22 +890,10 @@ describe('Optimistic recipient registry', () => {
       await provider.send('evm_increaseTime', [86400])
       await registry.executeRequest(recipientId)
 
-      await registry.removeRecipient(recipientId, { value: baseDeposit })
-      await expect(
-        registry.removeRecipient(recipientId, { value: baseDeposit })
-      ).to.be.revertedWith('RecipientRegistry: Request already submitted')
-    })
-
-    it('should not accept removal request with incorrect deposit size', async () => {
-      await registry.addRecipient(recipientAddress, metadata, {
-        value: baseDeposit,
-      })
-      await provider.send('evm_increaseTime', [86400])
-      await registry.executeRequest(recipientId)
-
-      await expect(
-        registry.removeRecipient(recipientId, { value: baseDeposit.div(2) })
-      ).to.be.revertedWith('RecipientRegistry: Incorrect deposit amount')
+      await registry.removeRecipient(recipientId)
+      await expect(registry.removeRecipient(recipientId)).to.be.revertedWith(
+        'RecipientRegistry: Request already submitted'
+      )
     })
 
     it('allows owner to challenge removal request', async () => {
@@ -906,12 +903,7 @@ describe('Optimistic recipient registry', () => {
       await provider.send('evm_increaseTime', [86400])
       await registry.executeRequest(recipientId)
 
-      await registry
-        .connect(requester)
-        .removeRecipient(recipientId, { value: baseDeposit })
-      const requesterBalanceBefore = await provider.getBalance(
-        requester.address
-      )
+      await registry.removeRecipient(recipientId)
       const requestRejected = await registry.challengeRequest(
         recipientId,
         requester.address
@@ -920,10 +912,6 @@ describe('Optimistic recipient registry', () => {
       expect(requestRejected)
         .to.emit(registry, 'RequestResolved')
         .withArgs(recipientId, RequestType.Removal, true, 0, currentTime)
-      const requesterBalanceAfter = await provider.getBalance(requester.address)
-      expect(requesterBalanceBefore.add(baseDeposit)).to.equal(
-        requesterBalanceAfter
-      )
 
       // Recipient is not removed
       expect(
@@ -942,14 +930,9 @@ describe('Optimistic recipient registry', () => {
       await provider.send('evm_increaseTime', [86400])
       await registry.executeRequest(recipientId)
 
-      await registry
-        .connect(requester)
-        .removeRecipient(recipientId, { value: baseDeposit })
+      await registry.removeRecipient(recipientId)
       await provider.send('evm_increaseTime', [86400])
 
-      const requesterBalanceBefore = await provider.getBalance(
-        requester.address
-      )
       const requestExecuted = await registry
         .connect(requester)
         .executeRequest(recipientId)
@@ -957,11 +940,6 @@ describe('Optimistic recipient registry', () => {
       expect(requestExecuted)
         .to.emit(registry, 'RequestResolved')
         .withArgs(recipientId, RequestType.Removal, false, 0, currentTime)
-      const txFee = await getTxFee(requestExecuted)
-      const requesterBalanceAfter = await provider.getBalance(requester.address)
-      expect(requesterBalanceBefore.sub(txFee).add(baseDeposit)).to.equal(
-        requesterBalanceAfter
-      )
 
       expect(
         await registry.getRecipientAddress(

